### DISCUSSION TITLE
[Next.js] How to change index (for localization)?

### DISCUSSION THREAD

**Discussion by JanKups** [algolia/instantsearch#5629] [Q&A] [answered]
I have multiple indices, each for a language (as [described](https://www.algolia.com/doc/guides/managing-results/optimize-search-results/handling-natural-languages-nlp/how-to/multilingual-search/#one-index-per-language)).

What would be the best approach to switch language in Next.JS? 

Options:
- Loading the index name as State and changing the State. Unfortunately this does not update the results.
- Sorting could be an option but seems illogical with different data

**Comment by dhayab** [comment, marked as answer]
You should be able to force a remount of `<InstantSearch>` by setting its `key` prop to your index name state getter.

```jsx
function Page() {
  const [indexName, setIndexName] = useState('...');
  
  /* ... */
  
  return (
    <InstantSearch searchClient={searchClient} indexName={indexName} key={indexName}>
      {/* ... */}
    </InstantSearch>
  );
}
```

**Reply by JanKups** [reply]
Awesome, works as mentioned!

**Reply by JanKups** [reply]
> You should be able to force a remount of `<InstantSearch>` by setting its `key` prop to your index name state getter.

@dhayab Any idea how to make this work with InstantSearchSSRProvider? I believe the server rendering is causing InstantSearch to not be refreshed.
